### PR TITLE
Adding an interval config to batch processor component

### PR DESCRIPTION
**Pull request by SachinVarghese**
**What this PR does / why we need it**:
This PR adds an interval functionality to the seldon batch processor component. This allows making batch predictions spread over time and is a very useful functionality to demonstate predictions made over time.

**Special notes for your reviewer**:

Seldon batch processor makes prediction in a multi-threaded format, so every worker will have this interval for predictions.

**Does this PR introduce a user-facing change?**:

```release-note
New time interval configuration for seldon batch processor
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **sachinvarghese**
You can assign the PR to them by writing `/assign @sachinvarghese` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by axsaucedo** [comment]
Interesting! Looks good - would be good to make sure to update the docs in the batch_processing page to mention this parameter as well. Also worth somehow documenting that the current setup only delays the adding to the queue by the time provided, which means that if the procesing time of a single data point takes 2 seconds with 1 second interval, there would be 2 requests at the same time at all times

File: python/seldon_core/batch_processor.py
Comment (RafalSkolasinski) [inline python/seldon_core/batch_processor.py:209]: could it be float? we may want to wait for 1.5 second :)

File: python/seldon_core/batch_processor.py
Comment (RafalSkolasinski) [inline python/seldon_core/batch_processor.py:667]: why change of the flag here? wouldn't `-b` work better?

File: python/seldon_core/batch_processor.py
Comment (SachinVarghese) [reply python/seldon_core/batch_processor.py:209]: Yeah good point, will update

File: python/seldon_core/batch_processor.py
Comment (RafalSkolasinski) [inline python/seldon_core/batch_processor.py:675]: `-t`?

File: python/seldon_core/batch_processor.py
Comment (SachinVarghese) [reply python/seldon_core/batch_processor.py:667]: batch-id was already using `-u`. Will that clash?

File: python/seldon_core/batch_processor.py
Comment (SachinVarghese) [reply python/seldon_core/batch_processor.py:675]: Yes `-t` makes sense will update

File: python/seldon_core/batch_processor.py
Comment (RafalSkolasinski) [reply python/seldon_core/batch_processor.py:667]: Oh yeah, it would clash :D (only when I expended code above I noticed). 
I'd suggest `-b` then.

Comment (RafalSkolasinski) [reply python/seldon_core/batch_processor.py:667]: Oh, `-b` also clash. Okay, then it's good.

**Review by RafalSkolasinski — CHANGES_REQUESTED** [review]
Nice one! Added few questions

File: python/seldon_core/batch_processor.py
Comment (SachinVarghese) [inline python/seldon_core/batch_processor.py:279]: @RafalSkolasinski I have updated the interval as the minimum time interval between batch prediction tasks

File: python/seldon_core/batch_processor.py
Comment (RafalSkolasinski) [inline python/seldon_core/batch_processor.py:276]: Later we could have default value of `batch_interval` to be `None` and here check 
```
if batch_interval is not None:
    ...
```

(but not sure if it makes sense to adjust now)

**Review by RafalSkolasinski — APPROVED** [review]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/3623#pullrequestreview-769062543 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3623]: [Method Not Allowed]

**Pull request merged**
